frontend/src/app/home/components/HomeView.js

```diff
@@ -34,6 +34,11 @@ export default class HomeView extends React.Component {
                     stocks={this.props.stocks}
                   />
                 </div>
+
+                 <Paper style={styles.ad}>
+                  <img src={this.props.ad}></img>
```

Comment:
I could be totally be wrong, but I don't think that resolving the advertisement via a promise is necessary.

Reply:
All that needs to be done is setting the src to the url that serves the ad.

Reply:
```
<Paper style={styles.ad}>
    <img src="/api/advertisement" alt="advertisement"></img>
</Paper> 
```